Source/WebCore/Modules/compression/CompressionStreamEncoder.cpp

```diff
@@ -114,7 +114,16 @@ ExceptionOr<RefPtr<JSC::ArrayBuffer>> CompressionStreamEncoder::compress(const u
     }
 
     while (shouldCompress) {
-        auto output = Vector<uint8_t>(allocateSize);
+        if (allocateSize < startingAllocationSize)
+            return Exception { TypeError, "Out of Memory."_s };
+
+        auto output = Vector<uint8_t>();
+        if (!output.tryReserveInitialCapacity(allocateSize)) {
+            allocateSize /= 4;
+            continue;
+        }
+
+        output.resize(allocateSize);
```

Comment:
Why output.resize(allocateSize), it seems unnecessary here given tryReserveInitialCapacity succeeded
storage.takeAsArrayBuffer() may return nullptr if not enough memory.
We should probably handle that case as well.
Something like:
```
auto result = storage.takeAsArrayBuffer();
if (!result)
     return Exception { OutOfMemoryError };
```

Reply:
The resize here should be just updating the internal m_size variable. This was needed to handle some behavior later with the code. I am not a huge fan of that line of code either. Will see if I can clean it up.

I fully agree on needing to check the result of `storage.takeAsArrayBuffer()`.